### PR TITLE
added convertion to julia's Dates.Time (function, doc string, docs, a…

### PR DESCRIPTION
**Pull request by yakir12**
…nd tests)

This came about because of [this discussion](https://discourse.julialang.org/t/why-do-time-quantities-have-to-be-integers/5864?u=yakir12) on Discourse.

### CODE REVIEW

**Comment by ajkeller34** [comment]
Do you have a use case for this? There's some uncertainty about the future of Base.Dates, and a lot of bike-shedding in the Discourse thread that's hard to follow. Code added here will cost me time to maintain in the future if anything changes in Base.Dates.

If you think this is a problem worth tackling, bidirectional conversion is a reasonable expectation. You can't convert back from a `Dates.CompoundPeriod` to Unitful quantities at present. I'd like to see that added before I consider merging.

I'm not sure if converting to `ns` first is a good idea. This code will fail with an overflow for integer-valued Unitful times above 293 years or so (e.g. `uconvert(Base.Dates.CompoundPeriod, 52*295u"wk")`). It might be better to convert to days first, subtract the integer number of days off, convert that to hours, etc., but I'm not sure. More strenuous tests would help find the corner cases.

Thanks for the PR.

**Comment by yakir12** [comment]
I see, fair enough. I'll close this now then, and hopefully the problem of using a float time will be solved somewhere else. Thanks for the heads-up about the overflow.
